crates/gpui/src/platform/windows/window.rs

```diff
@@ -109,75 +140,413 @@ impl WindowsWindowInner {
+        }
+    }
+
+    fn handle_move_msg(&self, lparam: LPARAM) -> LRESULT {
+        let x = lparam.loword() as f64;
+        let y = lparam.hiword() as f64;
+        self.origin.set(Point::new(x.into(), y.into()));
+        let mut callbacks = self.callbacks.borrow_mut();
+        if let Some(callback) = callbacks.moved.as_mut() {
+            callback()
+        }
+        LRESULT(0)
+    }
+
+    fn handle_size_msg(&self, lparam: LPARAM) -> LRESULT {
+        let width = lparam.loword().max(1) as f64;
+        let height = lparam.hiword().max(1) as f64;
+        self.renderer
+            .borrow_mut()
+            .update_drawable_size(Size { width, height });
+        let width = width.into();
+        let height = height.into();
+        self.size.set(Size { width, height });
+        let mut callbacks = self.callbacks.borrow_mut();
+        if let Some(callback) = callbacks.resize.as_mut() {
+            callback(
+                Size {
+                    width: Pixels(width.0),
+                    height: Pixels(height.0),
+                },
+                1.0,
+            )
+        }
+        LRESULT(0)
+    }
+
+    fn handle_paint_msg(&self) -> LRESULT {
+        let mut callbacks = self.callbacks.borrow_mut();
+        if let Some(callback) = callbacks.request_frame.as_mut() {
+            callback()
+        }
+        LRESULT(0)
+    }
+
+    fn handle_close_msg(&self, msg: u32, wparam: WPARAM, lparam: LPARAM) -> LRESULT {
+        let mut callbacks: std::cell::RefMut<'_, Callbacks> = self.callbacks.borrow_mut();
+        if let Some(callback) = callbacks.should_close.as_mut() {
+            if callback() {
+                return LRESULT(0);
+            }
+        }
+        drop(callbacks);
+        unsafe { DefWindowProcW(self.hwnd, msg, wparam, lparam) }
+    }
+
+    fn handle_destroy_msg(&self) -> LRESULT {
+        let mut callbacks: std::cell::RefMut<'_, Callbacks> = self.callbacks.borrow_mut();
+        if let Some(callback) = callbacks.close.take() {
+            callback()
+        }
+        let mut window_handles = self.platform_inner.window_handles.borrow_mut();
+        window_handles.remove(&self.handle);
+        if window_handles.is_empty() {
+            self.platform_inner
+                .foreground_executor
+                .spawn(async {
+                    unsafe { PostQuitMessage(0) };
+                })
+                .detach();
+        }
+        LRESULT(1)
+    }
+
+    fn handle_mouse_move_msg(&self, lparam: LPARAM, wparam: WPARAM) -> LRESULT {
+        let x = Pixels::from(lparam.loword() as f32);
+        let y = Pixels::from(lparam.hiword() as f32);
+        self.mouse_position.set(Point { x, y });
+        let mut callbacks: std::cell::RefMut<'_, Callbacks> = self.callbacks.borrow_mut();
+        if let Some(callback) = callbacks.input.as_mut() {
+            let pressed_button = match MODIFIERKEYS_FLAGS(wparam.loword() as u32) {
```

Comment:
is this really a `match`? looks like a series of `if/else` masked as a `match`

Reply:
yes, it is a series of if/else statements. not 'masking' it as a `match`. It just saves a temporary. I can make it a bunch of if statements if that's preferred, but I prefer the `match` in this case.